### PR TITLE
Preciso : Added new library to remove code duplication in bid adapter

### PR DESCRIPTION
**Pull request by NikhilGopalChennissery**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by patmmccann** [comment]
This still contains a bunch of tricks to avoid the report. Duplication still exists between Preciso utils and redtram and idx

```
      "fragment": "}\n\nfunction getBidFloor(bid) {\n  if (!isFn(bid.getFloor)) {\n    return deepAccess(bid, 'params.bidFloor', 0);\n  }\n\n  try {\n    const bidFloor = bid.getFloor({\n      currency: 'USD',\n      mediaType: '*',\n      size: '*',\n    });\n    return bidFloor.floor;\n  } catch (_) {\n    return 0\n  }\n}\n\nregisterBidder",
```

**Comment by patmmccann** [comment]
This feels like a hostile attempt to circumvent our guidelines

image

**Comment by NikhilGopalChennissery** [comment]
hi @patmmccann,  thank you for the updating me. 

we have changed bidfloor mapping logic in our library.

Thank You

**Comment by patmmccann** [comment]
Your pr is not yet reducing code duplication because it does not modify the redtram and idx bid adapters to import the library code

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann,

Do you mean that we need to modify the idx and redtram bid adapters? Do we have permission to change other adapters' code?

If we have permission to change other organizations' bidder adapters, we will import the library into those adapters as well.

As you suggested, we created a new library with the required functionality and modified the duplicated functionality. We then imported that library into our adapter to use its functionality.

I couldn't see any duplication in the library we created with other adapters.

Please correct me if I am wrong.

Thank you.

**Comment by patmmccann** [comment]
Yes, in the main branch you have duplication with those adapters, the goal of this to effort is to import common functions into all three.

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann,

I have imported and used a common library in all four adapters to remove the code duplication. Could you please check it when you have time?

Thank you.

File: libraries/bidUtils/bidUtils.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { convertOrtbRequestToProprietaryNative } from '../../src/native.js';
+import { replaceAuctionPrice } from '../../src/utils.js';
+import { getStorageManager } from '../../src/storageManager.js';
+import { ajax } from '../../src/ajax.js';
+import { MODULE_TYPE_UID } from '../../src/activities/modules.js';
+import { consentCheck } from './bidUtilsCommon.js';
+
+export const storage = getStorageManager({ moduleType: MODULE_TYPE_UID, moduleName: 'sharedId' });
```
Comment (patmmccann) [inline libraries/bidUtils/bidUtils.js:8]: Why is this module name chosen

File: libraries/bidUtils/bidUtilsCommon.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  isStr,
+  deepAccess,
+  getWindowTop,
+  triggerPixel
+} from '../../src/utils.js';
+import { BANNER, VIDEO, NATIVE } from '../../src/mediaTypes.js';
+import { getStorageManager } from '../../src/storageManager.js';
+import { MODULE_TYPE_UID } from '../../src/activities/modules.js';
+
+export const storage = getStorageManager({ moduleType: MODULE_TYPE_UID, moduleName: 'sharedId' });
```
Comment (patmmccann) [inline libraries/bidUtils/bidUtilsCommon.js:13]: Module name issue

**Comment by patmmccann** [comment]
Looking pretty good! almost at the finish line, some comments in slack

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann ,Thank you for your guidance.

I have made the changes. 
 
 Thanks a lot!

File: libraries/creative-renderer-display/renderer.js [hunk 3]
```diff
@@ -1,2 +1,2 @@
 // this file is autogenerated, see creative/README.md
-export const RENDERER = "!function(){\"use strict\";window.render=function({ad:d,adUrl:i,width:n,height:e},{mkFrame:o},r){if(!d&&!i)throw{reason:\"noAd\",message:\"Missing ad markup or URL\"};{const t=r.document,s={width:n,height:e};i&&!d?s.src=i:s.srcdoc=d,t.body.appendChild(o(t,s))}}}();"
\ No newline at end of file
+export const RENDERER = "(()=>{\"use strict\";window.render=function({ad:d,adUrl:e,width:i,height:r},{mkFrame:n},o){if(!d&&!e)throw{reason:\"noAd\",message:\"Missing ad markup or URL\"};{const s=o.document,t={width:i,height:r};e&&!d?t.src=e:t.srcdoc=d,s.body.appendChild(n(s,t))}}})();"
```
Comment (patmmccann) [inline libraries/creative-renderer-display/renderer.js:2]: lgtm, just undo your changes to this file and the below, these should be autogenerated and not modified by your pr

**Review by patmmccann — CHANGES_REQUESTED** [review]
code looks good, please dont modify the renderer.js files

File: libraries/creative-renderer-display/renderer.js (hunk 3 above)
Comment (NikhilGopalChennissery) [reply libraries/creative-renderer-display/renderer.js]: ooh sorry . I have reverted that changes.

**Comment by NikhilGopalChennissery** [comment]
> code looks good, please dont modify the renderer.js files

Ooh sorry. I have reverted that changes. 

Thank You!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**